config.yaml.sample

```diff
@@ -21,6 +21,9 @@ indexer:
   layer_scan_concurrency: 5
   migrations: true
 matcher:
+  matchers:
```

Comment:
@arajkumar 

Lets use the way we currently plug "Updater" config into clair as an example: https://quay.github.io/clair/reference/config.html#updaters-object

Can we emulate this same structure with a "Matcher" and "Config" subkey? 

It would be nice to keep things looking the say. @hdonnay please chime in asap if you do not agree.